### ISSUE TITLE
Contao symfony based version,doesn't show views in backend

### ISSUE THREAD

**Issue by hoss-ps** [contao/contao#399]
**Affected version(s)**
4.5 and 4.7.1
**Description**  
in contao backend , "Target' table" for "store form data" doesn't list mysql views .
in older version like 3.5 we could have it in the list.

**Comment by fritzmg** [comment]
Contao 4.5 has reached its end of life. Before reporting bugs, you must always first update to the most recent actively developed Contao version and test your bug there again. Currently that would be Contao 4.4.35 or 4.7.1.

**Comment by hoss-ps** [comment]
i have test in in version 4.7.1 and its the same.
does  it need to set some parameter for doctorin ?
seems in code bellow , **listTableNames**  wont return Views besid Tables
```php
	/**
	 * Return all tables as array
	 *
	 * @param string  $strDatabase The database name
	 * @param boolean $blnNoCache  If true, the cache will be bypassed
	 *
	 * @return array An array of table names
	 */
	public function listTables($strDatabase=null, $blnNoCache=false)
	{
		if ($blnNoCache || !isset($this->arrCache[$strDatabase]))
		{
			$strOldDatabase = $this->resConnection->getDatabase();

			// Change the database
			if ($strDatabase !== null && $strDatabase != $strOldDatabase)
			{
				$this->setDatabase($strDatabase);
			}

			$this->arrCache[$strDatabase] = $this->resConnection->getSchemaManager()->listTableNames();

			// Restore the database
			if ($strDatabase !== null && $strDatabase != $strOldDatabase)
			{
				$this->setDatabase($strOldDatabase);
			}
		}

		return $this->arrCache[$strDatabase];
	}
```

**Comment by leofeyer** [comment]
@contao/developers It seems this is an unnoticed BC break. Should we fix it and risk that some people have relied on the default Doctrine behaviour (`listTables()` separate from `listViews()`) or should we add it as known limitation?

**Comment by Toflar** [comment]
Maybe @hoss-ps can give us a bit more insight on why he needs the views to be present in the `listTables()` result?

**Comment by leofeyer** [comment]
Turns out this is a duplicate of contao/core-bundle#1250.

**Comment by leofeyer** [comment]
As discussed in Mumble on April 11th, we want to adjust the "store form data" select menu to also show views.

**Comment by xchs** [comment]
Shouldn't this be set to the 4.4.39 milestone?

**Comment by leofeyer** [comment]
Yes. Fixed.

**Comment by hoss-ps** [comment]
> Maybe @hoss-ps can give us a bit more insight on why he needs the views to be present in the `listTables()` result?

in case i need to access some tables of  3rd party database for same company which does not use any webservices and i have only limit access to some views of it .

in contao 3.x it works fine, and the issue is , once we update it to v4.x ,views exist in targetTable field in tl_form from previuse  version  but in backend we dont see it in table list, once we want to change it, errors appear.
it would be nice if combine listTable() and listViews() in backend table list.
by the way it can consider as updating bug since views store in targetTable  and deliver in version 4.x 
and cant change it in backend.

**Comment by leofeyer** [comment]
Fixed in 4874fc75ee5a96ed63e61efc3c5dddf4f00469aa.
